tests/nnstreamer_filter_flare/unittest_filter_flare.cc

```diff
@@ -0,0 +1,348 @@
+    GstBuffer *buffer;
+    GstMapInfo map;
+
+    data_size = strlen (data);
+    buffer = gst_buffer_new_allocate (NULL, data_size, NULL);
+
+    gst_buffer_map (buffer, &map, GST_MAP_WRITE);
+    memcpy (map.data, data, data_size);
+    gst_buffer_unmap (buffer, &map);
+
+    EXPECT_EQ (gst_app_src_push_buffer (GST_APP_SRC (appsrc), buffer), GST_FLOW_OK);
+  }
+};
+
+/**
+ * Note. Inference libraries like LLP and llama.cpp generally do not allow
+ * modification of such settings at runtime after the context (e.g., model,
+ * session, etc.) has been created. In particular, parameters like n_gpu_layers
+ * must be configured at model loading time and cannot be changed afterwards.
+ */
+
+/**
+ * @brief Test case for tensor_filter Flare plugin with singleInputSingleOutputSync_p
+ */
+TEST_F (NNStreamerFilterFlareCppTest, singleInputSingleOutputSync_p)
+{
+  gboolean invoke_async = FALSE;
+  gchar *contents = NULL;
+  gsize length = 0;
+  g_autofree gchar *custom = g_strdup_printf (
+      "tokenizer_path:%s,backend:CPU,output_size:1024,model_type:3B,data_type:W4A32,enable_fsu:false",
+      tokenizer);
+
+  if (skip_test) {
+    GTEST_SKIP () << "Skipping singleInputSingleOutputSync_p test due to missing model file";
+  }
+  ASSERT_TRUE (this->loaded);
+  new_sample_count = 0;
+
+  ASSERT_TRUE (g_file_get_contents (data_file, &contents, &length, NULL));
+
+  g_critical ("model: %s", model);
+  create_pipeline (model, invoke_async, custom);
+  data_push (contents);
+  /*need to lot's of time */
+  setPipelineStateSync (pipeline, GST_STATE_PLAYING, UNITTEST_STATECHANGE_TIMEOUT);
+  g_usleep (40000000);
+
+  EXPECT_EQ (setPipelineStateSync (pipeline, GST_STATE_NULL, UNITTEST_STATECHANGE_TIMEOUT), 0);
+  EXPECT_EQ (new_sample_count, 1);
+  g_free (contents);
+}
+
+/**
+ * @brief Test case for tensor_filter Flare plugin with multipleInputsSingleOutputSync_p
+ */
+TEST_F (NNStreamerFilterFlareCppTest, multipleInputsSingleOutputSync_p)
+{
+  gboolean invoke_async = FALSE;
+  gchar *contents = NULL;
+  gsize length = 0;
+  g_autofree gchar *custom = g_strdup_printf (
+      "tokenizer_path:%s,backend:CPU,output_size:1024,model_type:3B,data_type:W4A32,enable_fsu:false",
+      tokenizer);
+
+  if (skip_test) {
+    GTEST_SKIP () << "Skipping multipleInputsSingleOutputSync_p test due to missing model file";
+  }
+  ASSERT_TRUE (this->loaded);
+  new_sample_count = 0;
+
+  ASSERT_TRUE (g_file_get_contents (data_file, &contents, &length, NULL));
+
+  g_critical ("model: %s", model);
+  create_pipeline (model, invoke_async, custom);
+  data_push (contents);
+  data_push (contents);
+  /*need to lot's of time */
+  setPipelineStateSync (pipeline, GST_STATE_PLAYING, UNITTEST_STATECHANGE_TIMEOUT);
+  /*currently, API can support output_size. */
+  g_usleep (80000000);
+
+  EXPECT_EQ (setPipelineStateSync (pipeline, GST_STATE_NULL, UNITTEST_STATECHANGE_TIMEOUT), 0);
+  EXPECT_EQ (new_sample_count, 2);
+  g_free (contents);
+}
+
+#if 0 /*Flare call exit(0)*/
```

Comment:
If flare calls exit(0) somewhere in it, please report the code location of it. It should "throw", not "exit".